src/test/java/org/fogbowcloud/manager/core/ApplicationFacadeTest.java

```diff
@@ -12,1701 +12,2209 @@
+        Mockito.doReturn(order.getFederationUser()).when(this.aaaController)
+                .getFederationUser(Mockito.anyString());
 
-		String federationTokenValue = "";
-		this.application.deleteCompute(order.getId(), federationTokenValue);
+        Mockito.doNothing().when(this.aaaController).authenticate(Mockito.anyString());
 
-		Assert.assertEquals(OrderState.CLOSED, order.getOrderState());
-	}
+        Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
+                Mockito.any(Operation.class), Mockito.any(Order.class));
 
-	@Test
-	public void testDeleteComputeOrderUnathenticated() throws Exception {
-		Mockito.doThrow(new UnauthenticatedUserException()).when(this.aaaController).authenticate(Mockito.anyString());
+        // exercise
+        this.application.deleteCompute(order.getId(), FEDERATION_TOKEN_VALUE);
 
-		Order order = createComputeOrder();
-		OrderStateTransitioner.activateOrder(order);
+        // verify
+        Assert.assertEquals(OrderState.CLOSED, order.getOrderState());
+    }
+
+    // test case: When try calling the deleteCompute() method without user authentication, it must
+    // throw UnauthenticatedUserException and the Order remains in the same state.
+    @Test
+    public void testDeleteComputeOrderUnathenticated() throws Exception {
+
+        // set up
+        Order order = createComputeOrder();
+        OrderStateTransitioner.activateOrder(order);
 
-		Mockito.doReturn(order).when(this.orderController).getOrder(Mockito.anyString(),
-				Mockito.any(FederationUser.class), Mockito.any(InstanceType.class));
+        Mockito.doReturn(order).when(this.orderController).getOrder(Mockito.anyString(),
+                Mockito.any(FederationUser.class), Mockito.any(InstanceType.class));
 
-		Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
-				Mockito.any(Operation.class), Mockito.any(Order.class));
+        Mockito.doNothing().when(this.aaaController).authorize(Mockito.any(FederationUser.class),
+                Mockito.any(Operation.class), Mockito.any(Order.class));
+
+        Mockito.doThrow(new UnauthenticatedUserException()).when(this.aaaController)
+                .authenticate(Mockito.anyString());
+
+        try {
+            // exercise
+            this.application.deleteCompute(order.getId(), FEDERATION_TOKEN_VALUE);
+            Assert.fail();
+        } catch (UnauthenticatedUserException e) {
+            // verify
+            Assert.assertEquals(OrderState.OPEN, order.getOrderState());
+        }
+    }
 
-		try {
-			String federationTokenValue = "";
-			this.application.deleteCompute(order.getId(), federationTokenValue);
-			Assert.fail();
-		} catch (UnauthenticatedUserException e) {
-			Assert.assertEquals(OrderState.OPEN, order.getOrderState());
-		}
-	}
+    // test case: When try calling the deleteCompute() method without a token authentication, it
```

Comment:
I think we should improve the description and the code. It seems to me we are testing the facade behaviour when the controller.getFederationUse throws an exception. This is unrelated to "without a token authentication"